packages/react-button/etc/react-button.api.md

```diff
@@ -4,104 +4,70 @@
 
 ```ts
 
-import { ComponentPropsCompat } from '@fluentui/react-utilities';
-import type { ComponentStateCompat } from '@fluentui/react-utilities';
+import { ARIAButtonShorthandProps } from '@fluentui/react-aria';
+import type { ComponentProps } from '@fluentui/react-utilities';
+import type { ComponentState } from '@fluentui/react-utilities';
+import type { IntrinsicShorthandProps } from '@fluentui/react-utilities';
 import type { MenuTriggerChildProps } from '@fluentui/react-menu';
+import type { ObjectShorthandProps } from '@fluentui/react-utilities';
 import * as React_2 from 'react';
-import { ShorthandPropsCompat } from '@fluentui/react-utilities';
 
 // @public
-export const Button: React_2.FunctionComponent<ButtonProps & React_2.RefAttributes<HTMLElement>>;
+export const Button: React_2.ForwardRefExoticComponent<ButtonProps & React_2.RefAttributes<HTMLButtonElement>>;
 
 // @public (undocumented)
-export type ButtonDefaultedProps = 'icon' | 'size';
-
-// @public (undocumented)
-export type ButtonProps = ComponentPropsCompat & React_2.ButtonHTMLAttributes<HTMLElement> & {
+export type ButtonCommons = {
     appearance?: 'primary' | 'outline' | 'subtle' | 'transparent';
-    icon?: ShorthandPropsCompat<React_2.HTMLAttributes<HTMLElement>>;
     block?: boolean;
-    disabled?: boolean;
     disabledFocusable?: boolean;
+    disabled?: boolean;
     iconPosition?: 'before' | 'after';
-    shape?: 'rounded' | 'circular' | 'square';
-    size?: 'small' | 'medium' | 'large';
+    shape: 'rounded' | 'circular' | 'square';
```

Comment:
Making some optional and some not is confusing.  What is the right coding style for props and state for optionals?  Should every property in Commons be non-optional and then we always use Partial<XXXXCommons> for the props? (assuming every prop will be optional). If we have some props as optional then it seems like the State should make the ones it wants required.

IMO - the caller dev should be able to read the code such that the optional/required aspect is indicating the contract with the component (i.e. the props).

Reply:
We can do all of them required except the ones that need to be always optional, I'll make the change.